pydux/create_store.py

```diff
@@ -16,63 +16,55 @@ class ActionTypes(object):
     INIT = '@@redux/INIT'
 
 
-def create_store(reducer, initial_state=None, enhancer=None):
-    """
-    redux in a nutshell.
-
-    observable has been omitted.
-
-    Args:
-        reducer: root reducer function for the state tree
-        initial_state: optional initial state data
-        enhancer: optional enhancer function for middleware etc.
-
-    Returns:
-        a Pydux store
-    """
-    if enhancer is not None:
-        if not hasattr(enhancer, '__call__'):
-            raise TypeError('Expected the enhancer to be a function.')
-        return enhancer(create_store)(reducer)
-
-    if not hasattr(reducer, '__call__'):
-        raise TypeError('Expected the reducer to be a function.')
-
-    # single-element arrays for r/w closure
-    current_reducer = [reducer]
-    current_state = [initial_state]
-    current_listeners = [[]]
-    next_listeners = [current_listeners[0]]
-    is_dispatching = [False]
-
-    def ensure_can_mutate_next_listeners():
-        if next_listeners[0] == current_listeners[0]:
-            next_listeners[0] = current_listeners[0][:]
-
-    def get_state():
-        return current_state[0]
-
-    def subscribe(listener):
+class Store(object):
+    def __init__(self, reducer, initial_state, enhancer):
+        if enhancer is not None:
+            if not hasattr(enhancer, '__call__'):
+                raise TypeError('Expected the enhancer to be a function.')
+            return enhancer(create_store)(reducer)
+
+        if not hasattr(reducer, '__call__'):
+            raise TypeError('Expected the reducer to be a function.')
+
+        self.reducer = reducer
+        self.initial_state = initial_state
+        self.enhancer = enhancer
+        # single-element arrays for r/w closure
+        self.current_reducer = [reducer]
```

Comment:
All of these list[] wrappers and [0] indexes are for allowing read/write access to variables in closure.  If it is replaced with a class instance, than they can all go away.